### PR TITLE
Move validation to check if the prisoner's prison is supported on VSIP to the validate endpoint.

### PR DESCRIPTION
**Pull request by dhirajshettymoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PublicBookerController.kt [hunk 1]
```diff
@@ -171,7 +171,7 @@ class PublicBookerController(
       ApiResponse(
         responseCode = "422",
         description = "Prisoner validation failed",
-        content = [Content(mediaType = "application/json", schema = Schema(implementation = BookerPrisonerValidationException::class))],
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = PrisonerValidationErrorResponse::class))],
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/controller/PublicBookerController.kt:174]: good point, renamed to BookerPrisonerValidationErrorResponse

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PublicBookerService.kt [hunk 2]
```diff
@@ -58,9 +65,21 @@ class PublicBookerService(
   fun validatePrisoner(bookerReference: String, prisonerNumber: String) {
     logger.trace("validate prisoner called for $prisonerNumber with booker reference $bookerReference")
     prisonVisitBookerRegistryClient.validatePrisoner(bookerReference, prisonerNumber)
+
+    // if no errors that means prisoner's prison and registered prison matches
+    // check if the prisoner's prison is supported on Visits
+    prisonerSearchClient.getPrisonerById(prisonerNumber).prisonId?.let { prisonId ->
+      if (!isPrisonSupportedOnVisits(prisonId)) {
```
Comment (adaviesMOJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PublicBookerService.kt:72]: Does this belong in booker-registry as well? As it technically could be another returned 422 code validation error? 

Also is it possible for this to fail after validating the booker? how would their prison match but not be supported as we only store supported prison locations on the permitted prisoner table?

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PublicBookerService.kt:72]: the thinking is that the prison was enabled but for some reason later we decided to disable the prison for Public visits. A bit of an edge-case scenario.

Comment (adaviesMOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PublicBookerService.kt:72]: Would we still not want this validation in the same spot as all the other validation?

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PublicBookerService.kt:69]: as discussed, moved this check before the booker registry validation call

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**